File: Week 3/06. Turtle Pub & Sub/ros2_ws/src/pkg_cpp_py/src/talker.cpp
#include "rclcpp/rclcpp.hpp"
#include "pkg_cpp_py/talker.hpp"

using namespace std::chrono_literals;

Talker::Talker() : Node ("talker"){
    publisher_ = this->create_publisher<example_interfaces::msg::String>("chatter",10);
    timer_ = this->create_wall_timer(1s, std::bind(&Talker::talk, this));
    RCLCPP_INFO(this->get_logger(), "Talking...");
}

void Talker::talk(){
    auto msg = example_interfaces::msg::String();
    msg.data = std::string("Hi, I am ") + Talker::fullname_ + std::string(". I'm a Robotic Engineer.");
    publisher_->publish(msg);
}

int main(int argc, char** argv){
    rclcpp::init(argc, argv);
    rclcpp::spin(std::make_shared<Talker>());
    rclcpp::shutdown();
    return 0;
}

File: Week 3 & 4/06. Turtle Pub & Sub/ros2_ws/install/pkg_cpp_py/lib/pkg_cpp_py/talker.py
#!/usr/bin/env python3

import rclpy
from pkg_cpp_py.module_to_import import Talker

def main(args=None):
    rclpy.init(args=args)
    talker = Talker()
    rclpy.spin(talker)
    rclpy.shutdown()

if __name__ == '__main__':
    main()